File: SWEA/D2/2005. 파스칼의 삼각형/파스칼의 삼각형.py
t = int(input())
for tc in range(1,t+1):
    n = int(input())
    pascal = [[1]]
    for i in range(1,n):
        row =[1]
        for j in range(1,i):
            num = pascal[i-1][j-1]+pascal[i-1][j]
            row.append(num)
        row.append(1)
        pascal.append(row)
    print(f'#{tc}')
    for row in pascal:
        for num in row:
            print(num, end=' ')
        print()

File: 백준/Silver/16395. 파스칼의 삼각형/파스칼의 삼각형.java
import java.util.*;
import java.io.*;
import java.lang.reflect.Array;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        ArrayList<ArrayList<Integer>> pascal = new ArrayList<>();
        for(int i=0;i<n;i++){
            ArrayList<Integer>row = new ArrayList<>();
            for (int j = 0; j <= i; j++) {
                if (j == 0 || j == i) {
                    row.add(1); 
                } else {
                    
                    int value = pascal.get(i - 1).get(j - 1) + pascal.get(i - 1).get(j);
                    row.add(value);
                }
            }

            pascal.add(row);
        }

        
        System.out.println(pascal.get(n - 1).get(m - 1));
    }
}

